pkg/providers/v1/aws.go

```diff
@@ -3508,6 +3523,40 @@ func (c *Cloud) findSubnets() ([]*ec2.Subnet, error) {
 	return subnets, nil
 }
 
+// Returns a mapping between availability zone names and their types
+// Zone will not be included in the map in case it was not found in AWS by name
+func (c *Cloud) getZoneTypesByName(azNames []string) (map[string]string, error) {
+	if len(azNames) == 0 {
+		// if az names slice is empty, no need to make a request, return early with empty map
+		return map[string]string{}, nil
+	}
+	azFilter := newEc2Filter("zone-name", azNames...)
+	azRequest := &ec2.DescribeAvailabilityZonesInput{}
+	azRequest.Filters = []*ec2.Filter{azFilter}
+
+	azs, err := c.ec2.DescribeAvailabilityZones(azRequest)
+	if err != nil {
+		if awsErr, ok := err.(awserr.RequestFailure); ok && awsErr.Code() == "UnauthorizedOperation" {
```

Comment:
Tbh, if we ignore the error we will keep the existing behavior unless the IAM permissions get updated. I favor throwing the error here instead.

We will have to update the IAM permissions. As for EKS, we can start process to update the managed policies. For kops, we can request the permission to be included.

Reply:
Once the policies get updates, we can backport the changes.

Reply:
I updated pr, now permissions error will be propagated back. 

IAM permissions in kops were extended in https://github.com/kubernetes/kops/pull/14650.

@kishorj can you please take a look? :)